charts/datastore/templates/service.yaml

```diff
@@ -0,0 +1,24 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "datastore.fullname" . }}
+spec:
+  type: {{ .Values.service.type | default "ClusterIP" }}
+  {{- if and ((index .Values.service.loadBalancerIP) (eq .Values.service.type "LoadBalancer")) }}
+  loadBalancerIP: {{ .Values.service.loadBalancerIP }}
+  {{- end }}
+  selector:
+    product: {{ template "datastore.name" . }}  
+  ports:
+    {{ if (ne .Values.datastore.serverMode "TLS") }} # AUTOTLS mode requires port 80 for communicating with LetsEncrypt
```

Comment:
Maybe write this like:
`or eq "AUTOTLS" eq "PLAIN"`

instead of `ne "TLS"`, which is less clear (and will not do the right thing once we add a 4th mode)